rust/libceed/src/operator.rs

```diff
@@ -20,11 +20,218 @@
+    ///     q,
+    ///     "Incorrect field Basis"
+    /// );
+    /// assert_eq!(
+    ///     inputs[1].basis().unwrap().num_quadrature_points(),
+    ///     q,
+    ///     "Incorrect field Basis"
+    /// );
+    /// ```
+    pub fn basis(&self) -> Option<Basis> {
+        let mut ptr = std::ptr::null_mut();
+        unsafe {
+            bind_ceed::CeedOperatorFieldGetBasis(self.ptr, &mut ptr);
+        }
+        if ptr != unsafe { bind_ceed::CEED_BASIS_COLLOCATED } {
+            let mut ptr_copy = std::ptr::null_mut();
+            unsafe {
+                bind_ceed::CeedBasisReferenceCopy(ptr, &mut ptr_copy);
+            }
+            Some(Basis::from_raw(self.ceed, ptr_copy).unwrap())
+        } else {
+            None
+        }
+    }
+
+    /// Get the Vector of an OperatorField
+    ///
+    /// ```
+    /// # use libceed::prelude::*;
+    /// # let ceed = libceed::Ceed::default_init();
+    /// let qf = ceed.q_function_interior_by_name("Mass1DBuild").unwrap();
+    ///
+    /// // Operator field arguments
+    /// let ne = 3;
+    /// let q = 4 as usize;
+    /// let mut ind: Vec<i32> = vec![0; 2 * ne];
+    /// for i in 0..ne {
+    ///     ind[2 * i + 0] = i as i32;
+    ///     ind[2 * i + 1] = (i + 1) as i32;
+    /// }
+    /// let r = ceed
+    ///     .elem_restriction(ne, 2, 1, 1, ne + 1, MemType::Host, &ind)
+    ///     .unwrap();
+    /// let strides: [i32; 3] = [1, q as i32, q as i32];
+    /// let rq = ceed
+    ///     .strided_elem_restriction(ne, 2, 1, q * ne, strides)
+    ///     .unwrap();
+    ///
+    /// let b = ceed
+    ///     .basis_tensor_H1_Lagrange(1, 1, 2, q, QuadMode::Gauss)
+    ///     .unwrap();
+    ///
+    /// // Operator fields
+    /// let op = ceed
+    ///     .operator(&qf, QFunctionOpt::None, QFunctionOpt::None)
+    ///     .unwrap()
+    ///     .field("dx", &r, &b, VectorOpt::Active)
+    ///     .unwrap()
+    ///     .field("weights", ElemRestrictionOpt::None, &b, VectorOpt::None)
+    ///     .unwrap()
+    ///     .field("qdata", &rq, BasisOpt::Collocated, VectorOpt::Active)
+    ///     .unwrap();
+    ///
+    /// let inputs = op.inputs().unwrap();
+    ///
+    /// assert!(inputs[0].vector().is_none(), "Incorrect field Vector");
+    /// assert!(inputs[1].vector().is_none(), "Incorrect field Vector");
+    /// ```
+    pub fn vector(&self) -> Option<Vector> {
+        let mut ptr = std::ptr::null_mut();
+        unsafe {
+            bind_ceed::CeedOperatorFieldGetVector(self.ptr, &mut ptr);
+        }
+        if ptr != unsafe { bind_ceed::CEED_VECTOR_ACTIVE }
+            && ptr != unsafe { bind_ceed::CEED_VECTOR_NONE }
+        {
+            let mut ptr_copy = std::ptr::null_mut();
+            unsafe {
+                bind_ceed::CeedVectorReferenceCopy(ptr, &mut ptr_copy);
+            }
+            Some(Vector::from_raw(self.ceed, ptr_copy).unwrap())
+        } else {
+            None
+        }
+    }
+}
```

Comment:
It works, but I'm not really a fan of this because it does not distinguish between `CEED_VECTOR_NONE` and `CEED_VECTOR_ACTIVE`. Currently, we have the options enum `VectorOpt`, but that's technically more of a `VectorRefOpt`. I think we will want to have both an option enum for vectors and vector refs as well as a way to interop between the two.

Reply:
Ok, I have updated this with a better approach. If we like it, I can squash the two relevant commits together.